### PR TITLE
Update Jfrog docker registry URL

### PR DESCRIPTION
**Pull request by DropD**
## Description

Update JFrog artifactory URL.

Somehow the artifactory is not available anymore at the URL pointed to by the CSCS-CI organization's CSCS_REGISTRY variable. This breaks all CIGR pipelines in general. 
The existing credentials work however on the new location. The reason the new URL is hardcoded into the gitlab-ci.yml is that we can not change the externally managed variable.

## Requirements

Before submitting this PR, please make sure:

- [x] You have run the code checks, tests and documentation build successfully
- [x] All fixes and all new functionality are tested and documentation is up to date
- [x] You looked at the [review checklist](https://github.com/GridTools/gt4py/blob/master/REVIEW_CHECKLIST.txt)

Additionally, if this PR contains code authored by new contributors:

- [x] All the authors are covered by a valid contributor assignment agreement,
signed by the employer if needed, provided to ETH Zurich
- [x] The names of all the new contributors have been added to an updated
version of the AUTHORS.rst file included in the PR

### CODE REVIEW

**Comment by DropD** [comment]
bors try

**Comment by DropD** [comment]
bors try

**Comment by DropD** [comment]
bors try

**Comment by DropD** [comment]
bors try

**Comment by DropD** [comment]
bors retry

**Comment by DropD** [comment]
bors cancel

**Comment by DropD** [comment]
bors try

**Comment by DropD** [comment]
bors try

**Comment by DropD** [comment]
bors try

**Comment by havogt** [comment]
>The reason the new URL is hardcoded into the gitlab-ci.yml is that we can not change the externally managed variable.

Who is the "external" manager of these things? Should we ask to get this updated? Other people would also be affected or are we the last group on CIGR?

**Comment by gronerl** [comment]
I would be in favor of merging like this. At least we have an easy fix that is in our hands for next time this URL is changed. This will unblock other PR's so I'd appreciate if we don't have to wait for solving this with third parties.

**Review by gronerl — APPROVED** [review]

**Comment by havogt** [comment]
My question was not meant to block merging, but to make sure we do the right follow-up steps.

**Comment by gronerl** [comment]
I opened #953.

**Comment by gronerl** [comment]
Looks like CI is still failing, e.g. on master... is this the regular old occasional failure?

**Pull request merged**
